apps/shell/tash_command.c

```diff
@@ -638,18 +641,24 @@ bool tash_do_autocomplete(char *cmd, int *pos, bool double_tab)
 int tash_cmd_install(const char *str, TASH_CMD_CALLBACK cb, int thread_exec)
 {
 	int cmd_idx;
-	static int fail_cmd_count = 0;
-
-	if (TASH_MAX_COMMANDS == tash_cmds_info.count) {
-		printf("Allowed Max tash cmds: %d and Current tash cmd count: %d\n",
-				TASH_MAX_COMMANDS, tash_cmds_info.count + ++fail_cmd_count);
-		printf("Couldn't install cmd: (%s), Refer CONFIG_TASH_MAX_COMMANDS\n", str);
-		return -1;				/* MAX cmd count reached */
-	}
+	struct tash_cmd_s *new_cmd_buff;
 
 	/* Lock mutex */
 	pthread_mutex_lock(&tash_cmds_info.tmutex);
 
+	/* allocate the cmd buffer */
+	if (tash_cmds_info.count == 0) {
+		new_cmd_buff = (struct tash_cmd_s *)malloc(sizeof(struct tash_cmd_s));
+	} else {
+		new_cmd_buff = (struct tash_cmd_s *)realloc(tash_cmds_info.cmd, sizeof(struct tash_cmd_s) * (tash_cmds_info.count + 1));
+	}
+
+	if (new_cmd_buff) {
+		tash_cmds_info.cmd = new_cmd_buff;
+	} else {
+		return -1; /* Memory Allocation Fail for new command */
```

Comment:
How about printing the log we cannot install the cmd like before?

Reply:
Done